lib/media/ver10/set_video_encoder_configuration.ex

```diff
@@ -11,40 +11,136 @@ defmodule Onvif.Media.Ver10.SetVideoEncoderConfiguration do
   def request(device, args),
     do: Onvif.Media.Ver10.Media.request(device, args, __MODULE__)
 
+  def request_body(%VideoEncoderConfiguration{} = video_encoder_config)
+      when not is_nil(video_encoder_config.h264_configuration) and
+             not is_nil(video_encoder_config.mpeg4_configuration) do
+    element(:"s:Body", [
+      element(:"trt:SetVideoEncoderConfiguration", [
+        element(:"trt:Configuration", %{"token" => video_encoder_config.reference_token}, [
+          name_element(video_encoder_config),
+          use_count_element(video_encoder_config),
+          encoding_element(video_encoder_config),
+          quality_element(video_encoder_config),
+          resolution_element(video_encoder_config),
+          rate_control_element(video_encoder_config),
+          h264_element(video_encoder_config.h264_configuration),
+          mpeg4_element(video_encoder_config.mpeg4_configuration),
```

Comment:
can you have both in a valid configuration?

Reply:
or neither?

Reply:
also, probably a lot cleaner to have a `encoding_config_element/1` that returns the h264 configuration or mpeg4 configuration with the pattern matching there and then wrapping the Configuration list element in a `List.flatten/1` so it denests a list with multiple items or no items.

Reply:
Not sure if we can have both or neither since in the spec both are optional so I went will all combinations.

> also, probably a lot cleaner to have a encoding_config_element/1 that returns the h264 configuration or mpeg4 configuration with the pattern matching there and then wrapping the Configuration list element in a List.flatten/1 so it denests a list with multiple items or no items.

Okay

Reply:
If the encoding is JPEG both mpeg4 and h264 configurations are nil. So I think the possible combinations are

- h264, nil
- nil, mpeg4
- nil, nil

I am assuming having h264 and mpeg4 configurations is not a vaild one, shall I skip it?

Reply:
probably send an error back if they try both as invalid